java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java

```diff
@@ -649,7 +649,33 @@ public ResponseEntity<String> sendCode(@RequestBody EmailSenderInfo data) {
 		if (dev == null) {
 			return tokenNotValid();
 		}
-		return userdataService.sendCode(data.action, data.lang, dev);
+		PremiumUsersRepository.PremiumUser pu = usersRepository.findById(dev.userid);
+		if (pu == null) {
+			return ResponseEntity.badRequest().body("User not found");
+		}
+		return userdataService.sendCode(data.action, data.lang, pu);
+	}
+	
+	@PostMapping(path = {"/auth/send-code-to-new-email"})
```

Comment:
Should be protected against flooding PremiumUsersRepository with tons of emails.